src/core/xref.js

```diff
@@ -388,9 +388,10 @@ function addDataCiteToTerms(elems, queryKeys, data, conf) {
  * @param {any} conf
  */
 function addDataCite(elem, query, result, conf) {
-  const { term } = query;
-  const { uri, shortname: cite, normative, type, for: forContext } = result;
-
+  const { term, specs = [] } = query;
+  const { uri, shortname, spec, normative, type, for: forContext } = result;
+  // if authored spec context had `result.spec`, use it instead of shortname
+  const cite = specs.flat().includes(spec) ? spec : shortname;
```

Comment:
It seems `specs.flat()` happens frequently. Can it be flattened inside `getSpecContext`?

Reply:
The server relies on specContext being a nested array - first sub-array has highest priority and so on.

Reply:
What happens if everything is in the first subarray?

Reply:
This was done to avoid ambiguities with specContext. Consider:

``` html
<script>
  respecConfig.xref = ["fetch", "html"];
</script>
<body>
 <section data-cite="fetch dom"> 
   <a>body</a>
  <a>EventHandler</a> ...
 </section>
</body>
```

If we have everything in first specContext array, we have an ambiguity (body is defined in both fetch and html). So, we use a fallback chain, which goes like: search in [fetch, dom], if not found, search in [fetch, html]

Reply:
Understood, thanks. Although this should still be able to flattened first as this function is called after the query, but not sure how much it would improve the code.

Reply:
Would `query.specs.some(specs => specs.includes(spec)` be better than `flat()`?

Reply:
Not sure as it's significantly longer 🤔

Reply:
Though it avoids creating a new array..

Reply:
Well, shorter code is better than micro-optimization 👀